scripts/pylib/twister/harness.py

```diff
@@ -214,8 +215,15 @@ def pytest_run(self, log_file):
 class Test(Harness):
     RUN_PASSED = "PROJECT EXECUTION SUCCESSFUL"
     RUN_FAILED = "PROJECT EXECUTION FAILED"
+    test_suite_start_pattern = r"Running test suite (?P<suite_name>.*)"
 
     def handle(self, line):
+        test_suite_match = re.search(self.test_suite_start_pattern, line)
+        if test_suite_match:
+            suite_name = test_suite_match.group("suite_name")
+            self.detected_suite_names.append(suite_name)
+            self.ztest = True
```

Comment:
Maybe you can add a minor optimization here? I believe there can only be a single test suite name printed in a single run. We don't need to check each and every line for test suite's name. Maybe if it is once found the check can be skipped?

Reply:
Documentation allows to create multiple test suites in one test file:
https://docs.zephyrproject.org/latest/guides/test/ztest.html#quick-start-unit-testing
And indeed some tests have several test suites defined in one test file, like those:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/i2s/i2s_api/src/main.c
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bluetooth/controller/ctrl_data_length_update/src/main.c

So I cannot stop verify test suite name after find first one.